components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -68,7 +68,12 @@ export const rewardsPanelReducer: Reducer<RewardsExtension.State | undefined> =
         })
       } else if (result === RewardsExtension.Result.WALLET_CORRUPT) {
         state.walletCorrupted = true
-      } else if (result !== RewardsExtension.Result.LEDGER_OK) {
+      } else if (result === RewardsExtension.Result.LEDGER_OK) {
```

Comment:
This change fixes an issue where the panel will display the welcome UI after a migration, even when rewards is enabled.